app/main.py

```diff
@@ -1,6 +1,22 @@
 from typing import Callable
+import functools
+
+
+cached = {}
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    @functools.wraps(func)
+    def wrapper(*args) -> int:
+        global cached
+        key = str(id(func)) + "(" + f"{args}" + ")"
+        # print(key)
+        if key not in cached:
+            print("Calculating new result")
+            result = func(*args)
+            cached[key] = result
+        else:
+            print("Getting from cache")
+            result = cached[key]
+        return result
```

Comment:
Simplify this code.
In any case do `return cached[key]`

Write this in one line:
```
result = func(*args)
cached[key] = result
```